modules/t/Utils/FlatFile.t

```diff
@@ -24,15 +24,77 @@ use Test::Exception;
+    );
+
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, \@header1, '\t'), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
+
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2\t3\t4", "\t"), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
+
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, \@header2) } qr/Number of columns in header \(2\) do not match row \(4\)/, "Header doesn't match";
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2")  } qr/Number of columns in header \(2\) do not match row \(4\)/, "Header doesn't match";
+    # to test LIMIT = -1 of split function: https://perldoc.perl.org/functions/split
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2\t")  } qr/Number of columns in header \(3\) do not match row \(4\)/, "Header doesn't match";
+
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, \@header2, "\t") } qr/Number of columns in header \(2\) do not match row \(4\)/, "Header doesn't match";
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2", "\t") } qr/Number of columns in header \(2\) do not match row \(4\)/, "Header doesn't match";
+    # to test LIMIT = -1 of split function: https://perldoc.perl.org/functions/split
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2\t", "\t") } qr/Number of columns in header \(3\) do not match row \(4\)/, "Header doesn't match";
+
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, \@header2, '\t') } qr/Number of columns in header \(2\) do not match row \(4\)/, "Header doesn't match";
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2", '\t') } qr/Number of columns in header \(2\) do not match row \(4\)/, "Header doesn't match";
+    # to test LIMIT = -1 of split function: https://perldoc.perl.org/functions/split
+    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2\t", '\t') } qr/Number of columns in header \(3\) do not match row \(4\)/, "Header doesn't match";
+
+    
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_2, \@header1, ";"), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
+
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_2, "1;2;3;4", ";"), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
+
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_2, \@header1, ';'), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
+
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_2, "1;2;3;4", ';'), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
```

Comment:
As above, I'd suggest to reconsider the repetition of the test here with single- and double-quoted delimiter pattern strings. If it works for one, it should work for the other.